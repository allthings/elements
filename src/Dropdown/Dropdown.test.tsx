import React from 'react'
import { render } from '@testing-library/react'

import { Dropdown, ResourceProvider, ThemeProvider } from '../index'

describe('Dropdown', () => {
  it('should render with placement bottom (default)', () => {
    const { container } = render(
      <ThemeProvider>
        <ResourceProvider>
          <Dropdown
            initialSelectedItem={{
              label: 'dog',
              value: 'dog',
            }}
            label="some label"
            menuHeight={200}
            onSelect={() => {}}
            placeholder="select something..."
            clearable
            items={[
              {
                label: 'dog',
                value: 'dog',
              },
              {
                label: 'cat',
                value: 'cat',
              },
              {
                label: 'apple',
                value: 'apple',
              },
            ]}
          />
        </ResourceProvider>
      </ThemeProvider>,
    )
    expect(container).toMatchSnapshot()
  })
})
