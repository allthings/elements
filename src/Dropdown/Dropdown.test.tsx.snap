// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render with placement bottom (default) 1`] = `
.css-0,
[data-css-0] {
  position: absolute;
  left: 15px;
  font-size: 10px;
  opacity: 0;
  transition: all .225s ease-out;
  -webkit-transition: all .225s ease-out;
  -moz-transition: all .225s ease-out;
}

.css-1,
[data-css-1] {
  position: relative;
}

.css-2,
[data-css-2] {
  align-items: stretch;
  border: none;
  box-shadow: ;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex-direction: column;
  padding: 0;
  width: 100%;
  background-color: #ffffff;
  -webkit-box-align: stretch;
  -webkit-align-items: stretch;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -webkit-flex-direction: column;
}

.css-3,
[data-css-3] {
  width: 100%;
  height: 50px;
  position: relative;
  background-color: #ffffff;
}

.css-4,
[data-css-4] {
  width: calc(100% - 5px);
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  cursor: pointer;
  padding-right: 30px;
  color: black;
  background-color: white;
  pointer-events: none;
}

.css-5,
[data-css-5] {
  cursor: pointer;
  height: 100%;
  width: 50px;
  padding: 16px 16px 16px 20px;
}

.css-6,
[data-css-6] {
  box-sizing: border-box;
  flex: 0 0 auto;
  -webkit-flex: 0 0 auto;
}

.css-7,
[data-css-7] {
  box-sizing: border-box;
  flex: 0 0 auto;
  cursor: pointer;
  -webkit-flex: 0 0 auto;
}

.css-8,
[data-css-8] {
  display: inline;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 13px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  box-sizing: border-box;
  cursor: not-allowed;
  height: 50px;
  width: 100%;
  padding: 0 15px;
  padding-top: 10px;
  padding-right: 50px;
  transition: padding-top .225s ease-out;
  border: 0;
  -webkit-transition: padding-top .225s ease-out;
  -moz-transition: padding-top .225s ease-out;
}

.css-8:required,
[data-css-8]:required {
  box-shadow: none;
}

.css-8:-webkit-autofill ~ .label,
[data-css-8]:-webkit-autofill ~ .label {
  opacity: 1 !important;
  top: 8px !important;
}

.css-8:-webkit-autofill ~ .checkmark,
[data-css-8]:-webkit-autofill ~ .checkmark {
  opacity: 1 !important;
  top: 8px;
}

.css-8:-webkit-autofill,
[data-css-8]:-webkit-autofill {
  padding-top: 10px !important;
}

.css-8::-ms-clear,
[data-css-8]::-ms-clear {
  display: none;
}

.css-12,
[data-css-12] {
  position: absolute;
  top: 16px;
  right: 15px;
  pointer-events: none;
  transition: opacity .225s;
  opacity: 0;
  -webkit-transition: opacity .225s;
  -moz-transition: opacity .225s;
}

.css-13,
[data-css-13] {
  display: block;
  font-family: "Open Sans", Helvetica, Arial, sans-serif;
  font-weight: 400;
  font-size: 10px;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #626262;
}

.css-14,
[data-css-14] {
  width: 16px;
  height: 16px;
  fill: lightGrey;
}

.css-15,
[data-css-15] {
  position: absolute;
  top: 0;
  right: 0;
}

.css-16,
[data-css-16] {
  width: 10px;
  height: 10px;
  stroke: black;
}

<div>
  <div
    aria-expanded="false"
    aria-haspopup="listbox"
    aria-labelledby="downshift-0-label"
    data-css-2=""
    role="combobox"
  >
    <div
      data-css-1=""
      data-css-6=""
    >
      <div
        data-css-7=""
        data-css-3=""
      >
        <div
          data-css-1=""
          data-css-6=""
          style="width: 100%;"
        >
          <input
            aria-required="false"
            data-css-4=""
            data-css-8=""
            disabled=""
            placeholder="select something..."
            readonly=""
            type="text"
            value="dog"
          />
          <div
            class="label"
            data-css-6=""
            data-css-0=""
            style="opacity: 1; top: 8px;"
          >
            <div
              data-css-6=""
              data-css-13=""
            >
              some label
               
            </div>
          </div>
          <div
            class="checkmark"
            data-css-12=""
            data-css-6=""
          >
            <div
              data-css-6=""
              data-css-14=""
            />
          </div>
        </div>
        <div
          data-css-5=""
          data-css-7=""
          data-css-15=""
        >
          <div
            data-css-16=""
            data-css-6=""
            data-testid="dropdown-clear-icon"
          />
        </div>
      </div>
    </div>
    <div
      data-css-1=""
      data-css-6=""
    />
  </div>
</div>
`;
